tsl/src/continuous_aggs/refresh.c

```diff
@@ -879,9 +880,14 @@ continuous_agg_refresh_internal(const ContinuousAgg *cagg,
 
 	cagg = ts_continuous_agg_find_by_mat_hypertable_id(mat_id, false);
 
+	/* Cagg should be able to see tiered data despite the server settings */
+	ts_guc_enable_osm_reads = true;
```

Comment:
Hmmm... I don't think this way is the best, maybe we should introduce a GUC like `enable_cagg_osm_reads` that do this trick that is ON by default... wondering if for some reason (performance or whatever) we should NOT read osm data during the cagg refresh.